File: ifpe/teste2.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>

// Função para verificar se um número é primo
bool eh_primo(int n) {
    if (n <= 1) return false;
    if (n == 2) return true;
    if (n % 2 == 0) return false;
    for (int i = 3; i <= sqrt(n); i += 2) {
        if (n % i == 0) return false;
    }
    return true;
}

int main() {
    long int qnt_numeros;
    scanf("%d", &qnt_numeros);

    long long int num_recebe[qnt_numeros];
    for (int i = 0; i < qnt_numeros; i++) {
        scanf("%lld", &num_recebe[i]);
    }

    int resultados[1000][3]; // Armazena os resultados (número, j, k)
    int resultado_count = 0;

    for (int i = 0; i < qnt_numeros; i++) {
        int recebe = num_recebe[i];
        for (int j = 0; j < recebe; j++) {
            for (int k = 0; k < recebe; k++) {
                if ((j + k == recebe) && (j < k) && !eh_primo(j) && !eh_primo(k)) {
                    resultados[resultado_count][0] = recebe;
                    resultados[resultado_count][1] = j;
                    resultados[resultado_count][2] = k;
                    resultado_count++;
                }
            }
        }
    }

    for (int i = 0; i < qnt_numeros; i++) {
        long int calc_soma = 1000000000;
        bool apareceu = false;
        for (int j = 0; j < resultado_count; j++) {
            if (num_recebe[i] == resultados[j][0]) {
                if (resultados[j][2] - resultados[j][1] < calc_soma) {
                    calc_soma = resultados[j][2] - resultados[j][1];
                    apareceu = true;
                }
            }
        }

        if (!apareceu) {
            printf("-1\n");
        } else {
            for (int j = 0; j < resultado_count; j++) {
                if (num_recebe[i] == resultados[j][0]) {
                    if (resultados[j][2] - resultados[j][1] == calc_soma) {
                        printf("%d %d\n", resultados[j][1], resultados[j][2]);
                        break; // Para evitar múltiplas impressões
                    }
                }
            }
        }
    }

    return 0;
}

File: ifpe/teste2.py
def eh_primo(n):
    if n <= 1:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    for i in range(3, int(n**0.5) + 1, 2):
        if n % i == 0:
            return False
    return True



qnt_numeros = int(input())
resultados = []
num_recebe = []
for i in range(qnt_numeros):
    num_recebe.append(int(input()))

for recebe in num_recebe:
    for j in range(recebe):
        for k in range(recebe):
            if (j+k == recebe) and (j < k) and not eh_primo(j) and not eh_primo(k):
                resultados.append([recebe,j,k])

apareceu = False
for i in range(qnt_numeros):
    calc_soma = 1_000_000_000
    for j in range(len(resultados)):
        if num_recebe[i] == resultados[j][0]:
            calc_soma = min(calc_soma ,resultados[j][2]-resultados[j][1])
            apareceu = True

    if not apareceu:
        print(-1)
    for j in range(len(resultados)):
        if num_recebe[i] == resultados[j][0]:
            if resultados[j][2]-resultados[j][1] == calc_soma:
                print(resultados[j][1],resultados[j][2])
    
